packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -0,0 +1,203 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import { Encoding } from "src/libraries/Encoding.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { TransientContext, TransientReentrancyAware } from "src/libraries/TransientContext.sol";
+import { CrossL2Inbox } from "src/L2/CrossL2Inbox.sol";
+import { IL2ToL2CrossDomainMessenger } from "src/L2/IL2ToL2CrossDomainMessenger.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+
+/// @notice Thrown when a non-written slot in transient storage is attempted to be read from.
+error NotEntered();
+
+/// @notice Thrown when attempting to send a message to the chain that the message is being sent from.
+error MessageDestinationSameChain();
+
+/// @notice Thrown when attempting to relay a message and the function caller (msg.sender) is not CrossL2Inbox.
+error RelayMessageCallerNotCrossL2Inbox();
+
+/// @notice Thrown when attempting to relay a message where CrossL2Inbox's origin is not L2ToL2CrossDomainMessenger.
+error CrossL2InboxOriginNotL2ToL2CrossDomainMessenger();
+
+/// @notice Thrown when attempting to relay a message whose destination chain is not the chain relaying it.
+error MessageDestinationNotRelayChain();
+
+/// @notice Thrown when attempting to relay a message whose target is CrossL2Inbox.
+error MessageTargetCrossL2Inbox();
+
+/// @notice Thrown when attempting to relay a message whose target is L2ToL2CrossDomainMessenger.
+error MessageTargetL2ToL2CrossDomainMessenger();
+
+/// @notice Thrown when attempting to relay a message that has already been relayed.
+error MessageAlreadyRelayed();
+
+/// @custom:proxied
+/// @custom:predeploy 0x4200000000000000000000000000000000000023
+/// @title L2ToL2CrossDomainMessenger
+/// @notice The L2ToL2CrossDomainMessenger is a higher level abstraction on top of the CrossL2Inbox that provides
+///         features necessary for secure transfers ERC20 tokens between L2 chains. Messages sent through the
+///         L2ToL2CrossDomainMessenger on the source chain receive both replay protection as well as domain binding.
+contract L2ToL2CrossDomainMessenger is IL2ToL2CrossDomainMessenger, ISemver, TransientReentrancyAware {
+    /// @notice Transient storage slot that `entered` is stored at.
+    ///         Equal to bytes32(uint256(keccak256("crossl2inbox.entered")) - 1)
+    bytes32 internal constant ENTERED_SLOT = 0x6705f1f7a14e02595ec471f99cf251f123c2b0258ceb26554fcae9056c389a51;
```

Comment:
I dont think we need entered slot anymore

Reply:
good call!